bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -1031,8 +988,11 @@ public void deleteSnapshots(String regex) throws IOException {
    */
   @Override
   public void deleteSnapshots(Pattern pattern) throws IOException {
-    for (SnapshotDescription snapshotDescription : listSnapshots(pattern)) {
-      deleteSnapshot(snapshotDescription.getName());
+    if (pattern != null && !pattern.matcher("").matches()) {
```

Comment:
@liubonan PTAL at this bit - in previous versions this was unimplemented so I made use of the listSnapshots API as the previous implementation did

Reply:
I personally think this should be left as Unimplemented. `deleteSnapshots` sounds to be an atomic operation while it actually multiple API calls - one to list backups and N to delete individual backups. This may lead to unexpected results.

Reply:
the hbase impl is not atomic, so I think this is ok